pkg/image/api/helper.go

```diff
@@ -165,6 +165,21 @@ func (r DockerImageReference) String() string {
 	return fmt.Sprintf("%s%s%s", registry, r.Namespace, r.NameString())
 }
 
+// StringNoTag converts a DockerImageReference to a Docker pull spec with no tag.
+func (r DockerImageReference) StringNoTag() string {
+	registry := r.Registry
+	if len(registry) > 0 {
+		registry += "/"
+	}
+
+	if len(r.Namespace) == 0 {
```

Comment:
why not use a local var like you do for registry above? You're not modifying the ref because you're working on a copy of it, but if in the future we change the method receiver to a ptr, then you'd be inadvertently mutating the object by just getting its String